hypershift-operator/controllers/nodepool/conditions.go

```diff
@@ -308,11 +308,19 @@ func (r *NodePoolReconciler) validArchPlatformCondition(ctx context.Context, nod
 }
 
 func (r *NodePoolReconciler) validMachineConfigCondition(ctx context.Context, nodePool *hyperv1.NodePool, hcluster *hyperv1.HostedCluster) (*ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
 	releaseImage, err := r.getReleaseImage(ctx, hcluster, nodePool.Status.Version, nodePool.Spec.Release.Image)
 	if err != nil {
 		return &ctrl.Result{}, fmt.Errorf("failed to look up release image metadata: %w", err)
 	}
 
+	// TODO (alberto): we should hide haproxy config generation within NewConfigGenerator and let eht kubeconfig ausence to be an error type.
```

Comment:
```suggestion
	// TODO (alberto): we should hide haproxy config generation within NewConfigGenerator and let the kubeconfig absence to be an error type.
```

Reply:
Feel free to ignore. I just report what I see.

Reply:
hahaha, you beat me to it 😄